types/applepayjs/index.d.ts

```diff
@@ -1035,6 +1059,46 @@ declare namespace ApplePayJS {
         newTotal: ApplePayLineItem;
     }
 
+    /**
+     * The attributes contained by the oncouponcodechanged callback function.
+     */
+    abstract class ApplePayCouponCodeUpdate {
+        /**
+         * An optional list of line items.
+         */
+        newLineItems?: ApplePayLineItem[] | undefined;
+
+        /**
+         * The new total resulting from a change in the payment method.
+         */
+        newTotal: ApplePayLineItem;
+
+        /**
+         * An array of updated multitoken contexts for a multimerchant payment request.
+         */
+        newMultiTokenContexts?: ApplePayPaymentTokenContext[];
+
+        /**
+         * An updated request for an automatic reload payment.
+         */
+        newAutomaticReloadPaymentRequest?: ApplePayAutomaticReloadPaymentRequest;
+
+        /**
+         * An updated request for a recurring payment.
+         */
+        newRecurringPaymentRequest?: ApplePayRecurringPaymentRequest;
```

Comment:
```suggestion
        /**
         * An updated request for a recurring payment.
         */
        newRecurringPaymentRequest?: ApplePayRecurringPaymentRequest;

        /**
         * An updated request for a deferred payment.
         */
        newDeferredPaymentRequest?: ApplePayDeferredPaymentRequest;
```

Reply:
Hey, I have added that new type `ApplePayDeferredPaymentRequest`. I got the descriptions from the [documentation](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaydeferredpaymentrequest).

But I believe there is something weird with the descriptions of **freeCancellationDate** and **freeCancellationDateTimeZone**. Both have the description "The time zone at the destination location of the payment."  🤔

I'm not familiar with those fields, do you know more about them or have a different idea?

Reply:
I've not used them myself - from looking at the documentation I assumed that `Date freeCancellationDate` is the local time for when you can cancel for free until, and `DOMString freeCancellationDateTimeZone` is the IANA time zone ID that local date is in, e.g. `new Date()` and `Europe/London` for where I am right now.